six/projects/csm/tests/test_sicd_csm.cpp

```diff
@@ -0,0 +1,362 @@
+
+            six::RowColDouble doubleImagePoint(
+                    static_cast<double>(scpPixel.row),
+                    static_cast<double>(scpPixel.col));
+
+            six::RowColDouble imagePointDifference =
+                    convertedImagePoint - doubleImagePoint;
+            imagePointDifference.row = std::abs(imagePointDifference.row);
+            imagePointDifference.col = std::abs(imagePointDifference.col);
+            imageDifferences[ii] = std::max(imagePointDifference.row,
+                    imagePointDifference.col);
+
+            six::RowColInt adjustedScpPixel(
+                    scpPixel.row - mComplexData->imageData->firstRow,
+                    scpPixel.col - mComplexData->imageData->firstCol);
+
+            scene::Vector3 convertedGroundPoint =
+                    imageToGround(*model, adjustedScpPixel, height,
+                    offsets[ii]);
+            scene::Vector3 groundPointDifference =
+                    convertedGroundPoint - scp.ecf;
+
+            for (size_t jj = 0; jj < 3; ++jj)
+            {
+                groundPointDifference[jj] = std::abs(groundPointDifference[jj]);
+            }
+            groundDifferences[ii] = std::max(groundPointDifference[0],
+                    std::max(groundPointDifference[1],
+                            groundPointDifference[2]));
+        }
+
+        double leastGroundDifference = *std::min_element(
+                groundDifferences.begin(), groundDifferences.end());
+        double leastImageDifference = *std::min_element(
+                imageDifferences.begin(), imageDifferences.end());
+
+        if (leastGroundDifference != groundDifferences[0] ||
+            leastImageDifference != imageDifferences[0])
+        {
+            std::cerr << "There was an offset better than " <<
+                    offsets[0] << "\n";
+            testPassed = false;
+        }
+
+        if (leastGroundDifference > tolerance)
+        {
+            std::cerr << "Converted ground point > " << tolerance <<
+                    " away from SCP\n";
+            testPassed = false;
+        }
+
+        if (leastImageDifference > tolerance)
+        {
+            std::cerr << "Converted image point > " << tolerance <<
+                    " away from scpPixel\n";
+        }
+        return testPassed;
+    }
+
+private:
+    static const char MODEL_NAME[];;
+
+    const std::string mSicdPathname;
+    const csm::Plugin& mPlugin;
+
+    six::XMLControlRegistry mXmlRegistry;
+    six::NITFReadControl mReader;
+    std::auto_ptr<six::sicd::ComplexData> mComplexData;
+};
+
+std::string findDllPathname(const std::string& installPathname)
+{
+    const std::string csmPluginPathname =
+            sys::Path(installPathname).join("share").join("CSM").join("plugins");
+    const std::vector<std::string> csmPlugins =
+            sys::Path::list(csmPluginPathname);
+
+    // TODO: If multiple plugins, which is correct?
```

Comment:
I'd add this function to your shared functions that both test programs can use too